crate/server/src/routes/google_cse/operations.rs

```diff
@@ -193,7 +220,261 @@ pub async fn unwrap(
     // re-extract the bytes from the key
     let dek = wrapped_dek.key_block()?.key_bytes()?;
 
+    debug!("unwrap: exiting with success");
     Ok(UnwrapResponse {
         key: general_purpose::STANDARD.encode(dek),
     })
 }
+
+/// Request to perform a private key signature.
+/// The `digest` is signed using unwrapped `wrapped_private_key`,
+/// and using `algorithm`.
+///
+/// Technical specifications of components from this request
+/// can be found here: https://support.google.com/a/answer/7300887
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PrivateKeySignRequest {
+    pub authentication: String,
+    pub authorization: String,
+    /// The algorithm that was used to encrypt the Data Encryption Key (DEK) in envelope encryption.
+    pub algorithm: String,
+    /// Base64-encoded message digest.
+    /// The digest of the DER encoded SignedAttributes.
+    /// This value is unpadded. Max size: 128B
+    pub digest: String,
+
+    /// The format of the private key or the wrapped private key is up to
+    /// the Key Access Control List Service (KACLS) implementation.
+    /// On the client and on the Gmail side, this is treated as an opaque blob.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub e_key: Option<String>,
```

Comment:
remove this param ?

Reply:
done in a83e9638f1de6b310f771e776498c958c1d9f7d6